Python/tier2_redundancy_eliminator_bytecodes.c

```diff
@@ -197,6 +197,17 @@ dummy_func(void) {
         }
     }
 
+    op(_BINARY_OP_ADD_UNICODE, (left, right -- res)) {
+        if (is_const(left) && is_const(right)) {
+            PyObject *temp = PyUnicode_Concat(get_const(left), get_const(right));
+            ERROR_IF(temp == NULL, error);
+            OUT_OF_SPACE_IF_NULL(res = sym_new_const(ctx, temp));
```

Comment:
I think we know how to do this right now (see gh-116062). Unless you'd first like to do a pure cleanup pass that introduces `ERROR_IF` (and maybe `BOTTOM_IF` as well?)?